eval/testdata/eval/unterminated-array/expected.json

```diff
@@ -9,7 +9,7 @@
             "Expression": null,
             "EvalContext": null,
             "Extra": null,
-            "Shown": false
+            "Path": ""
```

Comment:
Is this right?

Reply:
I think it is because it is unterminated we could only provide a partial path

Reply:
I think Levi is right, wouldnt we want the path for this?

Reply:
This error happens during [yaml unmarshaling](https://github.com/pulumi/esc/blob/c5d9a8a4b4ee56955ea6e386378b83fb75d48c40/syntax/encoding/yaml.go#L403-L405), before we do our parsing from yaml to esc AST. That's why there also isn't a `Subject (no range)` in this error.